File: tree.js
export default class Node {
    constructor(value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }

    insert(value) {
        if (value < this.value) {
            if (this.left === null) {
                this.left = new Node(value);
            } else {
                this.left.insert(value);
            }
        } else if (value > this.value) {
            if (this.right === null) {
                this.right = new Node(value);
            } else {
                this.right.insert(value)
            }
        }
    }

    print(padding = '', pipe = false, pointer = '   ') {

        console.log(padding + pointer + this.value)

        if (pipe)
            padding += '│  ';
        else
            padding += '   ';
            
        if (this.left !== null) {
            this.left.print(
                padding,
                (this.right ? true : false),
                (this.right ? '├──' : '└──')
            );
        }
        if (this.right !== null) {
            this.right.print(padding, false, '└──');
        }
    }
}

File: tree.py
# binary tree node class
class Node:
    def __init__(self, value) -> None:
        self.value = value
        self.left = None
        self.right = None

    def insert(self, value):
        if value < self.value:
            if self.left is None:
                self.left = Node(value)
            else:
                self.left.insert(value)
        elif value > self.value:
            if self.right is None:
                self.right = Node(value)
            else:
                self.right.insert(value)
    
    def print(self, padding = '', pipe = False, pointer = '   '):

        print(padding + pointer + str(self.value))

        if pipe:
            padding += '│  '
        else:
            padding += '   '

        if self.left is not None:
            self.left.print(padding, True if self.right else False, '├──' if self.right else '└──')

        if self.right is not None:
            self.right.print(padding, False, '└──')
